expfmt/text_parse.go

```diff
@@ -287,6 +296,35 @@ func (p *TextParser) startLabelName() stateFn {
 		p.parseError(fmt.Sprintf("invalid label name for metric %q", p.currentMF.GetName()))
 		return nil
 	}
+	if p.skipBlankTabIfCurrentBlankTab(); p.err != nil {
+		return nil // Unexpected end of input.
+	}
+	if p.currentByte != '=' {
+		if p.currentMetricIsInsideBraces {
+			switch p.currentByte {
+			case ',':
+				p.setOrCreateCurrentMF()
+				p.currentMetric = &dto.Metric{}
+				return p.startLabelName
+			case '}':
+				p.setOrCreateCurrentMF()
+				p.currentMetric = &dto.Metric{}
+				p.currentMetric.Label = append(p.currentMetric.Label, p.currentLabel...)
+				p.currentLabel = nil
+				if p.skipBlankTab(); p.err != nil {
+					return nil // Unexpected end of input.
+				}
+				return p.readingValue
+			default:
+				p.parseError(fmt.Sprintf("unexpected end of metric value %q", p.currentByte))
+				return nil
+			}
+		} else {
```

Comment:
All code paths in the `if` clause above end up returning, so we don't need the `else` here.